torch/linalg/__init__.py

```diff
@@ -367,6 +367,91 @@
+
+Computes the least squares solution to the system with a batch of matrices :math:`a` (represented by :attr:`input`)
+and a batch of vectors/matrices :math:`b` such that
+
+.. math::
+    x = \text{argmin}_x \|ax - b\|_F,
+
+where :math:`a` is of size :math:`(..., m, n)` and :math:`b` is of size :math:`(..., m, k)` or
+:math:`(..., m)`. The batch dimensions of :math:`a` and :math:`b` have to be broadcastable.
+
+The returned solution :math:`solution` is of shape :math:`(..., n, k)` if :math:`b` is of shape :math:`(..., m, k)`,
+and is of shape :math:`(..., n, 1)` if :math:`b` is of shape :math:`(..., m)`.
+The batch sizes of :math:`x` is the broadcasted shape of the batch dimensions of :math:`a` and :math:`b`.
+
+.. note::
+    The case when :math:`m < n` is not supported on the GPU yet.
+
+Args:
+    input (Tensor): the batch of left-hand side matrices :math:`a`
+        of shape :math:`(..., m, n)` with :math:`m > 0, n > 0`
+    b (Tensor): the batch of righ-hand side vectors or matrices :math:`b`
+        of shape :math:`(..., m)` or :math:`(..., m, k)` with :math:`m > 0, k > 0`
+    cond (float, optional): used to determine the effective rank of :math:`a`
+        for the rank-revealing drivers (see :attr:`driver`).
+        Singular values :math:`s[i] \le cond * s[0]` are treated as zero.
+        If :attr:`cond` is ``None`` or is smaller than zero,
+        the machine precision based on ``a.dtype`` is used.
+        Default: ``None``
+    driver (str, optional): the name of the LAPACK/MAGMA driver that is used
+        to compute the solution.
+        If the inputs are on the CPU, the valid values are
+        (``'gels'``, ``'gelsy'``, ``'gelsd``, ``'gelss'``, ``None``).
+        If the inptus are on the GPU, the valid values are (``'gels'``, ``None``).
+        If ``driver == None``, ``'gelsy'`` is used for CPU inputs and ``'gels'`` for GPU inputs.
+        Default: ``None``
+
+.. note::
+    Driver ``'gels'`` assumes only full-rank inputs, i.e. ``torch.matrix_rank(a) == min(m, n)``.
+    Drivers ``'gelsy'``, ``'gelsd'``, ``'gelss'`` are rank-revealing and hence handle rank-deficient inputs.
+    ``'gelsy'`` is QR-based with column pivoting, and ``'gelsd'`` with ``'gelss'`` are SVD-based.
+    ``'gelsy'`` is the fastest among the rank-revealing algorithms that also handles rank-deficient inputs.
+
+Returns:
+    (Tensor, Tensor, Tensor, Tensor): a namedtuple (solution, residuals, rank, singular_values) containing:
+        - **solution** (*Tensor*): the least squares solution
+        - **residuals** (*Tensor*):  if :math:`m > n` then for full rank matrices in :math:`a` the tensor encodes
+            the squared residuals of the solutions, that is :math:`||ax - b||_F^2`.
+            If :math:`m <= n`, an empty tensor is returned instead.
+        - **rank** (*Tensor*): the tensor of ranks of the matrix :math:`a` with shape ``a.shape[:-2]``.
+            Non-empty if :attr:`driver` is one of (``'gelsy'``, ``'gelsd'``, ``'gelss'``).
```

Comment:
I would rephrase this positively as "Only computed if the driver is one of ..., an empty tensor is returned otherwise."